compiler/rustc_parse/src/parser/path.rs

```diff
@@ -484,7 +492,9 @@ impl<'a> Parser<'a> {
         let arg = self.parse_generic_arg(ty_generics)?;
         match arg {
             Some(arg) => {
-                if self.check(&token::Colon) | self.check(&token::Eq) {
+                let separated =
+                    self.check_noexpect(&token::Colon) || self.check_noexpect(&token::Eq);
+                if separated && (self.check(&token::Colon) | self.check(&token::Eq)) {
```

Comment:
instead of using `check` here, is there a method that just adds the token to the list of expected tokens? Maybe we could use that instead, if so.